src/transformStateWithClones.js

```diff
@@ -7,7 +7,30 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const resultArray = [];
+
+  for (let i = 0; i < actions.length; i++) {
+    const data = actions[i];
+
+    if (data.type === 'addProperties') {
+      Object.assign(state, data.extraData);
```

Comment:
You are directly modifying the original state object. Instead, create a copy of the state object to avoid mutations. Use a proper naming convention for the copy, such as `stateCopy`. [CODE STYLE]